File: python/0003_longest_substring_without_repeating_characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        string = list(s)
        hashmap = []
        length = 0
        
        for i in string:
            if i not in hashmap:
                hashmap.append(i)
                temp = len(hashmap)
                if temp > length:
                    length = temp
            else:
                idx = hashmap.index(i)
                hashmap = hashmap[idx+1:]
                hashmap.append(i)
                
        return length

File: cpp/0003_longest_substring_without_repeating_characters.cpp
#include <iostream>
#include <string>
#include <vector>

using namespace std;

class Solution {
 public:
  int lengthOfLongestSubstring(string s) {
    int len_substring = 0, counter = 0;
    int length_string = s.length();
    vector<char> seen;

    for (char c : s) {
      bool present = false;
      for (auto i : seen) {
        if (i == c) present = true;
        counter++;
      }
      if (present) {
        auto X = seen.size() - counter + 1;
        vector<int> result(X);

      } else {
        counter += 1;
        seen[c] = 1;
      }
    }

    return len_substring;
  }
};

int main() {
  Solution s;
  int length = s.lengthOfLongestSubstring("hello");
  cout << length << endl;
}
